hs/Database/Tables.hs

```diff
@@ -183,29 +162,19 @@ data Course =
 instance ToJSON Course
 instance ToJSON Session
 
-instance ToJSON Lecture where
-  toJSON (Lecture lectureExtra lectureSection lectureCap lectureTimeStr lectureTime lectureInstructor lectureEnrol lectureWait)
-          = object ["extra" .= lectureExtra,
-                    "section" .= lectureSection,
-                    "cap" .= lectureCap,
-                    "time_str" .= lectureTimeStr,
-                    "time" .= map convertTimeToString lectureTime,
-                    "instructor" .= lectureInstructor,
-                    "enrol" .= lectureEnrol,
-                    "wait" .= lectureWait
-                   ]
-
-instance ToJSON Tutorial where
-  toJSON (Tutorial Nothing tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
-  toJSON (Tutorial (Just value) tutorialTimes tutorialTimeStr) =
-      Array $ V.fromList [toJSON value, toJSON (map convertTimeToString tutorialTimes), toJSON tutorialTimeStr]
+instance ToJSON Time where
+    toJSON (time) =
+        toJSON $ convertTimeToString time
+
+-- instance FromJSON required so that tables can be parsed into JSON,
+-- not necessary otherwise.
+instance FromJSON Time
```

Comment:
Summary of discussion:

- Define new `Enum` instance `DayOfWeek`, and make it an instance of `toJSON` and `fromJSON`
- Change the `Time` type to be a record containing two fields, `day :: DayOfWeek` and `time :: TimeOfDay`
- We should then be able to use the generic `json` to handle `Time`

Javascript files will also need to be changed to handle the new json format, but overall everything should be cleaner.